### PR TITLE
Adding external lib to do reverse geocoding; displaying obtained addr…

### PR DESCRIPTION
**Pull request by RobinCsl**
…esses; setting destination marker on click

Please, check also code from https://github.com/kiwicom/react-native-reverse-geocode

### CODE REVIEW

File: app/MMB/package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "@kiwicom/mobile-pdf": "^0",
     "@kiwicom/mobile-relay": "^0",
     "@kiwicom/mobile-shared": "^0",
+    "@kiwicom/react-native-reverse-geocode": "git+ssh://git@github.com:kiwicom/react-native-reverse-geocode.git#master",
```
Comment (tbergq) [inline app/MMB/package.json]: I think we should deploy it to npm and include it from there.

Comment (RobinCsl) [reply app/MMB/package.json]: But I guess, I really want the guys to have a look at it before publishing it. Or maybe it does not matter?

File: app/MMB/src/navigation/AddressPickerScreen.js [hunk 2]
```diff
@@ -95,23 +102,41 @@ export class AddressPickerScreen extends React.Component<Props, State> {
...
+  };
+
   render() {
+    const searchText =
+      idx(this.props.navigation, _ => _.state.params.searchText) || '';
+    const region = idx(this.props.navigation, _ => _.state.params.region);
     return (
-      <ScrollView>
+      <ScrollView keyboardShouldPersistTaps="handled">
         {this.state.formattedAddress != '' && (
```
Comment (tbergq) [inline app/MMB/src/navigation/AddressPickerScreen.js]: I know it is not part of this PR, but this should probably have been `!== ''`

Comment (RobinCsl) [reply app/MMB/src/navigation/AddressPickerScreen.js]: Absolutely, will fix it.

File: app/MMB/src/scenes/tripServices/transportation/Address.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+export default class Address extends React.Component<Props> {
+  setLocation = () => {
+    this.props.setLocation(this.props.address.location);
+  };
+  render() {
+    return (
+      <Touchable onPress={this.setLocation}>
+        <View style={styles.wrapper}>
+          {Platform.OS === 'ios' &&
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/transportation/Address.js]: Will there be some code for android here also?

File: app/MMB/src/scenes/tripServices/transportation/Addresses.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      return (
+        <View>
+          <Text style={styles.header}>
+            <Translation id="mmb.trip_service.transportation.address_picker.all_places" />
+          </Text>
+          {this.state.addresses.map((address, index) => (
+            <Address
+              setLocation={this.props.setLocation}
+              address={address}
+              key={address.address + index}
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/transportation/Addresses.js]: `${address.address}${index}`

File: app/MMB/src/scenes/tripServices/transportation/Address.js (hunk 3 above)
Comment (RobinCsl) [reply app/MMB/src/scenes/tripServices/transportation/Address.js]: Last time I could check on an Android phone, there was no title being displayed, only the address.

File: app/MMB/package.json (hunk 1 above)
Comment (tbergq) [reply app/MMB/package.json]: You could always publish it to npm, and if it requires changes, you can publish a new version. But yeah, when @ferrannp and @lukewalczak are happy with the code, you can publish to npm. And then we can merge this pr.

**Comment by ferrannp** [comment]
Linking code looks good to me.

**Comment by RobinCsl** [comment]
NPM package published and included in this PR.
@tbergq 🚀

File: app/MMB/src/navigation/AddressPickerScreen.js [hunk 5]
```diff
@@ -95,23 +102,41 @@ export class AddressPickerScreen extends React.Component<Props, State> {
...
-                <Translation id="mmb.trip_service.transportation.address_picker.current_location_title" />
-              </Text>
-              <Text style={styles.currentLocationAddress}>
-                <Translation passThrough={this.state.formattedAddress} />
-              </Text>
+      <ScrollView keyboardShouldPersistTaps="handled">
+        {this.state.formattedAddress !== '' && (
+          <Touchable onPress={this.setLocationToCurrentLocation}>
+            <View style={styles.currentLocationContainer}>
+              <TextIcon code="&quot;" style={styles.currentLocationIcon} />
```
Comment (tbergq) [inline app/MMB/src/navigation/AddressPickerScreen.js]: Could you use orbit icons, since I have added support for that now?

Comment (RobinCsl) [reply app/MMB/src/navigation/AddressPickerScreen.js]: Absolutely

File: app/MMB/src/navigation/AddressPickerScreen.js [hunk 6]
```diff
@@ -95,23 +102,41 @@ export class AddressPickerScreen extends React.Component<Props, State> {
...
-          <View style={styles.currentLocationContainer}>
-            <TextIcon code="&quot;" style={styles.currentLocationIcon} />
-            <View>
-              <Text style={styles.currentLocationTitle}>
-                <Translation id="mmb.trip_service.transportation.address_picker.current_location_title" />
-              </Text>
-              <Text style={styles.currentLocationAddress}>
-                <Translation passThrough={this.state.formattedAddress} />
-              </Text>
+      <ScrollView keyboardShouldPersistTaps="handled">
```
Comment (tbergq) [inline app/MMB/src/navigation/AddressPickerScreen.js]: Keyboard should normally close on scroll, and also you should use Keyboard avoiding view.

**Pull request merged**